vpr/src/route/route_timing.cpp

```diff
@@ -850,24 +886,39 @@ timing_driven_route_structs::~timing_driven_route_structs() {
                                      rt_node_of_sink);
 }
 
-void reduce_budgets_if_congested(route_budgets& budgeting_inf,
-                                 CBRR& connections_inf,
-                                 float slope,
+void reduce_budgets_if_congested(std::vector<ClusterNetId>& rerouted_nets,
+                                 route_budgets& budgeting_inf,
                                  int itry) {
-    auto& cluster_ctx = g_vpr_ctx.mutable_clustering();
-    if (budgeting_inf.if_set()) {
-        for (auto net_id : cluster_ctx.clb_nlist.nets()) {
-            if (should_route_net(net_id, connections_inf, false)) {
+    (void)rerouted_nets;
+    if (budgeting_inf.if_set() && itry > 20) {
+        // for (auto net_id : rerouted_nets) {
+        //     if (budgeting_inf.get_should_reroute(net_id)) {
+        //         budgeting_inf.update_congestion_times(net_id);
+        //     } else {
+        //         budgeting_inf.not_congested_this_iteration(net_id);
+        //     }
+        // }
+
+        /*Problematic if the overuse nodes are positive or declining at a slow rate
+         * Must be more than 9 iterations to have a valid slope*/
+        // if (slope > CONGESTED_SLOPE_VAL && itry >= 9) {
+        // VTR_LOG("Lowering budgets\n");
+        // budgeting_inf.lower_budgets(1.2);
+        // }
+    }
+}
+
+void increase_short_path_crit_if_congested(std::vector<ClusterNetId>& rerouted_nets,
+                                           route_budgets& budgeting_inf,
+                                           int itry) {
+    if (budgeting_inf.if_set() && itry > 9) {
+        for (auto net_id : rerouted_nets) {
+            if (budgeting_inf.get_should_reroute(net_id)) {
                 budgeting_inf.update_congestion_times(net_id);
             } else {
                 budgeting_inf.not_congested_this_iteration(net_id);
             }
-        }
-
-        /*Problematic if the overuse nodes are positive or declining at a slow rate
-         * Must be more than 9 iterations to have a valid slope*/
-        if (slope > CONGESTED_SLOPE_VAL && itry >= 9) {
-            budgeting_inf.lower_budgets(1e-9);
+            budgeting_inf.increase_short_crit(net_id, 4);
```

Comment:
Why 4?  Why 9?  Please change this magic numbers to named constants, with comments explaining the intention behind them.

Reply:
This is another function it might be better to delete entirely, I was experimenting with it earlier but haven't found it useful. I left it in because it was already in the route_timing code.